src/patternfly/components/Masthead/masthead.scss

```diff
@@ -135,35 +150,34 @@ $pf-v6-c-masthead--inset-map: build-spacer-map("none", "sm", "md", "lg", "xl", "
   }
 }
 
-// - Masthead main masthead content masthead logo
-.#{$masthead},
 .#{$masthead}__main {
-  position: relative;
+  display: var(--#{$masthead}__main--Display);
+  column-gap: var(--#{$masthead}__main--ColumnGap);
 }
 
-.#{$masthead}__main,
-.#{$masthead}__content,
-.#{$masthead}__logo,
+// - Masthead main masthead content masthead logo
+.#{$masthead},
 .#{$masthead}__brand {
-  display: flex;
-  align-self: stretch;
+  position: relative;
 }
 
-// - Masthead main masthead brand masthead content
-.#{$masthead}__main,
+// - Masthead brand, logo, content
 .#{$masthead}__brand,
+.#{$masthead}__logo,
 .#{$masthead}__content {
+  display: flex;
+  align-self: stretch;
```

Comment:
I might change this to `align-self: center` just for `__logo`. This will center the image relative to the toggle and make `__logo` hug the image instead of potentially being taller than the image if the image is shorter than `__logo`

before (with the logo height adjusted to 25px instead of 37px)
<img width="394" alt="Screenshot 2024-07-17 at 1 37 41 PM" src="https://github.com/user-attachments/assets/e28cb671-0322-4937-992e-9b406e2d0c8b">

after
<img width="378" alt="Screenshot 2024-07-17 at 1 39 10 PM" src="https://github.com/user-attachments/assets/6ee4aa30-e9ca-4eb7-9e9a-3e8264b9127b">